plugins/ercc-vscc/ercc_validation_logic.go

```diff
@@ -232,13 +236,14 @@ func policyErr(err error) *commonerrors.VSCCEndorsementPolicyError {
 }
 
 func getEccNamespace(txRWSet *rwsetutil.TxRwSet) (string, error) {
-	// Readsets must contain only one ecc_*.MRENCLAVE
+	// Readsets must contain only one *some_fpc_chaincode_name*.MRENCLAVE
 	eccNamespace := ""
 	for _, ns := range txRWSet.NsRwSets {
 		for _, read := range ns.KvRwSet.Reads {
-			if read.Key == MrEnclaveStateKey && strings.HasPrefix(ns.NameSpace, "ecc_") {
+			if read.Key == MrEnclaveStateKey {
+				logger.Debugf("found MRENCLAVE within namespace: %s", ns.NameSpace)
 				if eccNamespace != "" {
-					return "", fmt.Errorf("mutiple namespaces for ecc found")
+					return "", fmt.Errorf("mutiple namespaces with MRENCLAVE key found")
 				}
 				eccNamespace = ns.NameSpace
```

Comment:
I guess the current code would allow wrap an ecc _setup-tx by (via cc2cc) by another tx and "hijack" the attestation and register it with a different ecc_namespace?  Per-se i don't see immediately how this leads to bad stuff but it's a bit ugly.  Ideally, i think the attestation should explicitly bind the the ecc namespace?  (Not necessarily here, but might be good to do so in the revised version? Actually, iff we implement the new registration protocol, we should do so ... :-)

Reply:
I linked this conversation to #134